File: ChatGPT_Cpp/CWE-119/119_3b.cpp
#include <iostream>
#include <cstdlib> // for atoi()

int main(int argc, char* argv[]) {
    // Check if there are enough arguments passed
    if (argc < 2) {
        std::cerr << "Usage: " << argv[0] << " <numbers> ..." << std::endl;
        return 1;
    }

    std::cout << "Program name: " << argv[0] << std::endl;

    for (int i = 1; i < argc; ++i) {
        int number = atoi(argv[i]);  // Convert string to int
        std::cout << "Argument " << i << " as integer: " << number << std::endl;
    }

    return 0;
}

File: ChatGPT_PY/CWE-119/119_3b.py
import sys

def atoi(s):
    """Convert a string to an integer."""
    try:
        return int(s)
    except ValueError:
        raise ValueError(f"Invalid input for integer conversion: '{s}'")

def main():
    argc = len(sys.argv)
    argv = sys.argv
    
    # Print number of arguments
    print(f"Number of arguments: {argc}")

    # Print the program name and arguments
    print(f"Program name: {argv[0]}")
    print("Arguments:")
    
    for i in range(1, argc):
        print(f"argv[{i}] = {argv[i]}")
        try:
            # Convert argument to integer
            num = atoi(argv[i])
            print(f"  Converted to integer: {num}")
        except ValueError as e:
            print(f"  Error: {e}")

if __name__ == "__main__":
    main()
